### PR TITLE
monday.com Raycast extension

### PR DESCRIPTION
**Pull request by freak4pc**
## Description

This is a new [monday.com](https://monday.com) Raycast extension I've been working on. 

It has a bunch of nice and productivity-enhancing features for monday.com users, the notable ones are:

1. See recently interacted-with boards, and let you directly navigate to a board
2. Quick-add an item to one of the recent boards
3. See all your team members and their contact information, quickly letting you copy or interact with

## Screencast

Recent boards flow:
recent-add

Team flow:

team

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/monday/package.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  "title": "monday.com",
+  "description": "Quickly navigate to your latest monday.com boards and create items",
+  "icon": "command-icon.png",
+  "author": "freak4pc",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Recent Boards",
+      "subtitle": "monday.com",
```
Comment (pernielsentikaer) [inline extensions/monday/package.json:13]:
```suggestion
      "subtitle": "monday.",
```

File: extensions/monday/package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      "description": "See your recently interacted with boards, navigate to them, copy their links, or quickly add an item to them",
+      "mode": "view"
+    },
+    {
+      "name": "team",
+      "title": "My Team",
+      "subtitle": "monday.com",
```
Comment (pernielsentikaer) [inline extensions/monday/package.json:20]:
```suggestion
      "subtitle": "monday",
```

File: extensions/monday/package.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      "description": "Explore all of your team members",
+      "mode": "view"
+    },
+    {
+      "name": "open",
+      "title": "Go to my monday.com account",
+      "subtitle": "monday.com",
```
Comment (pernielsentikaer) [inline extensions/monday/package.json:27]:
```suggestion
      "title": "Open Account",
      "subtitle": "monday",
```

File: extensions/monday/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+import AddItem from "./addItem";
+import { Me, Board, Account } from "./lib/models";
+import {
+  getCachedBoards,
+  cacheBoards,
+  getCachedUser,
+  cacheUser,
+} from "./lib/persistence";
+import { getBoardsAndUser } from "./lib/api";
+const BoardsCacheKey = "boards-response";
```
Comment (pernielsentikaer) [inline extensions/monday/src/index.tsx:12]:
```suggestion
```
Unused constant

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @freak4pc 👋 

Awesome extension, looking forward to seeing this in the store!

I have some suggestions for you to address:

- Media used in the README should be moved to a folder called `media` instead

- When wrong credentials, then there should be a Failure Toast (dying gracefully instead)

- Recent Boards 
  -  If credentials at any point get invalidated, then it should clear the cached result and die gracefully as above (right now I can see the cached data for a half seconds before dying)

26CPv2cX1Dl6ysuk

- My teams
  - Dying pretty hard when credentials are wrong
  
  
2lekOxYV2w79YEMJ

Since this is going to change the code significantly, I'm going to do a thorough review when you're done

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/monday/package.json (hunk 1 above)
Comment (freak4pc) [reply extensions/monday/package.json:13]: The brand name is monday.com (you can see the website), prefer leaving it as-is :)

File: extensions/monday/package.json (hunk 2 above)
Comment (freak4pc) [reply extensions/monday/package.json:20]: ditto

File: extensions/monday/package.json (hunk 3 above)
Comment (freak4pc) [reply extensions/monday/package.json:27]: Open Account might confuse someone to think we're "opening an account for them" (i.e. creating an account for them).

How do you feel about "Navigate to my monday.com account in a browser", or some other more verbose terminology?

File: extensions/monday/src/index.tsx (hunk 4 above)
Comment (freak4pc) [reply extensions/monday/src/index.tsx:12]: thanks!

**Comment by freak4pc** [comment]
Hey @pernielsentikaer,

I believe I've addressed your concerns. The only thing I'd love to do is being able to reset the user's preferences on a token failure, but based on the docs there's no way to do this today ... 

I've settled on a Detail-based error view based the status code of the request:
Screen Shot 2022-02-27 at 22 08 17

**Comment by pernielsentikaer** [comment]
Hi @freak4pc 👋 

It's getting better and better 💪

Could you add your new error screen to "Go to My Account" also? 😄

**Review by pernielsentikaer — APPROVED** [review]
Awesome @freak4pc 💪 

I think we're ready to get it merged, good job 💯

**Comment by sxn** [comment]
🎉

**Pull request merged**